### PR TITLE
How to develop an API architecture

### LINKED ISSUE #4735: [API] How to develop an API architecture
#### Proposed title of article
How to develop an API architecture

#### Proposed article introduction
Industries have recently experienced enormous growth in the use of mobile applications that are frequently communicating with the backend over the network or web. This has no doubt made use of APIs an important business resource that any organization cannot overlook.
As a result through the use of APIs, businesses have recorded massive digital success. Over the past few years, huge multinational businesses have been using APIs to increase and diversify their channels enhancing the digital revolution. A number of SMEs are also adopting the trend.
So the biggest dilemma arises for business owners on how to develop APIs that suit their needs. The developers have to use specific API architecture which leverages on which API to use for the business. This article will expound more on the same.

#### Key takeaways 
At the end of the article the learner will understand:
- the concept of APIs and their purposes
- what is meant by API architecture and its components
- the core practices and objectives of designing an API architecture
- the key architectural styles of designing the API and several layers of API architecture

#### Article quality
Most of the articles found online are much concentrated with specific API architectural design which does not cover what each developer may require compared to this article which covers several designs

#### References
N/A

### PR DESCRIPTION
**Pull request by kerubo-tech**
> ### Article checklist
> #### Pre-submission checks
> > Please ensure that you have only one open topic suggestion (issue) + in-review article (linked pull request) at a time. This will ensure that we complete the review process in a timely manner from inception to publishing
> 
> > If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.
> 
> > For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.
> 
> #### Formatting and structure checks
> * [x]  Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
> * [x]  Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
> * [x]  Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
> * [x]  Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
> * [x]  Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
> * [x]  Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
> * [x]  Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
> * [x]  Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
> * [x]  Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
> 
> #### Grammar & plagiarism checks
> * [x]  Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com)
> * [x]  Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
> * [x]  Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
> * [x]  Have you added sources for quotes and images that aren't yours?
> 
> #### Technical checks
> * [x]  Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
> * [x]  Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
> * [x]  Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?
> 
> #### Topic suggestion this closes
> > Remove the backticks and add the issue number below to link and close your Topic Suggestion (issue) when your article has been published (PR has been merged). See this [video](https://youtu.be/TKJ4RdhyB5Y?t=278) for more details.
> 
> This closes #4735
> 
> > Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Plagiarism check done.

File: content/articles/how-to-develop-an-api-architecture/index.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+- API design goals should be aligned with the goals of the given business
+- The required resources have to be available, and proper management ensured
+- The use cases supported by the API should be considered
+- The target audience  and users should be predefined
+- The API value addition, channels, and ensure developers’ motivation to encourage its usage
+- The applications intended to make use of the developed API should be beneficial to the end-users
+
+#### Best practices when designing an API
+The following best practices should be considered when designing an API:
+- An API should draw its focus on its consumption and ensure consistency
```
Comment (mbaabuones) [inline content/articles/how-to-develop-an-api-architecture/index.md:72]: Put full stops

File: content/articles/how-to-develop-an-api-architecture/index.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+The API should have a consistent standard that directs each application interaction. There should be consistency when exchanging essential resources in a given institution, enhancing transparency in all the operations.
+
+#### Security
+An API should have an extra security layer that ensures all the requests are not directly made to the server. This is important since it ensures there is minimal chance for a breach or unauthorized access to the server.
+
+#### Scalability
+All organizations strive to increase and expand their operations. This would mean that the API should be designed to allow scalability to handle a growth in the products offered, security needs, or even an increase in data needs.
+
+#### Collaboration
+API should ensure components' reusability in application development.  This saves development time and prevents the developers from performing redundant and repetitive tasks repeatedly. This also assists the developers in focusing on trying and testing new tools, which is more valuable to the organization. It also helps the developers hire third-party companies to handle more complex tasks as they focus on the organization system.
```
Comment (mbaabuones) [inline content/articles/how-to-develop-an-api-architecture/index.md:91]: delete the word 'repeatedly'

**Review by mbaabuones — CHANGES_REQUESTED** [review]
@kerubo-tech Great article! The content is rich and well-written. Please work on the suggested changes to improve the article. After revising the article, go through the article to check and correct grammar errors. Ping me when you are done.

**Comment by hectorkambow** [comment]
@kerubo-tech

It seems this PR is still quite a bit aways from being completed. 

We will be closing this and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer.

Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue. 👍 Be sure to consult previously published materials to get a better understanding of the level of quality the community expects.

Please feel free to go through our provided resources as well - those are there to help all our students improve their overall technical writing. To be able to take those skills to any position or program.

**Pull request closed without merging**